### PR TITLE
Add Thycotic SecretServer support

### PR DESCRIPTION
**Pull request by EndlessTrax**
##### SUMMARY

Add a Credential Plugin for Thycotic SecretServer (TSS)

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API
 - UI

##### AWX VERSION

```
awx: 19.2.1
```

### CODE REVIEW

**Comment by EndlessTrax** [comment]
Sorry for the additional commits. This PR should be good now.

**Comment by shanemcd** [comment]
Hello. We recently switched to using GitHub Actions. Can you please rebase on top of the devel branch so tests run here?

**Review by jbradberry — APPROVED** [review]

File: awx/main/credential_plugins/tss.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+tss_plugin = CredentialPlugin(
+    'Thycotic Secret Server',
+    tss_inputs,
+    lambda **kwargs: ServerSecret(
+        **SecretServer(kwargs['server_url'], PasswordGrantAuthorizer(kwargs['server_url'], kwargs['username'], kwargs['password'])).get_secret(
+            kwargs['secret_id']
+        )
+    ).fields[kwargs['secret_field']],
+)
```
Comment (jakemcdermott) [inline awx/main/credential_plugins/tss.py]: I don't see a `lambda` as necessary here. In my opinion, readability would be greatly improved and complexity reduced by moving the callable to its own function.

More importantly, I have concerns about debugging or modifying the code. A common thing for maintainers to do with these plugins is to embed a [breakpoint](https://docs.python.org/3/library/pdb.html#pdb.set_trace) within the data request logic; inlining the whole backend as a lambda precludes that. 

```suggestion
def tss_backend(**kwargs):
    tss_authorizer = PasswordGrantAuthorizer(kwargs['server_url'], kwargs['username'], kwargs['password'])
    tss_server = SecretServer(kwargs['server_url'], tss_authorizer)
    tss_secret = tss_server.get_secret(kwargs['secret_id'])

    return ServerSecret(**tss_secret).fields[kwargs['secret_field']]


tss_plugin = CredentialPlugin(
    'Thycotic Secret Server',
    inputs=tss_inputs,
    backend=tss_backend
)
```

**Review by jakemcdermott — CHANGES_REQUESTED** [review]

**Comment by amigus** [comment]
I generally only create functions (or variables) when I need them because my brain has been damaged from too much functional programming. Given that and the fact that the function is one line and delegates to the SDK API, which is what you end up debugging anyway, I figured a lambda was fine.

That said, regardless of my reasoning, I concede that it's harder to debug, so I'm just going to follow the convention and push the change you requested momentarily.

**Comment by jakemcdermott** [comment]
@amigus Sounds good. You'll want to rebase the branch with latest devel and force-push, then we'll be ready to merge.

**Comment by amigus** [comment]
Could someone else us understand how our change (adding a credential plugin) can break the UI? I have no idea how to debug this. Sorry. 😞

**Comment by amigus** [comment]
My bad. My repo didn't have the latest upstream. I just rebased and force-pushed. Hopefully gtg now.

**Review by jakemcdermott — APPROVED** [review]

**Comment by amigus** [comment]
Thanks for your support everyone.

**Pull request merged**